src/comotion/dash.py

```diff
@@ -215,6 +220,30 @@ def generate_presigned_url_for_file_upload(self, file_key: str = None) -> FileUp
 
         return self.load_api_instance.generate_presigned_url_for_file_upload(self.load_id, file_upload_request=file_upload_request)
         
+    def upload_file(self, file: Union[str, io.FileIO], file_upload_response: FileUploadResponse):
```

Comment:
add file_upload response as an optional. IF not supplied then automatically fetch.  Supply filekey as an optional.  Also, boolen option to use the file name as filekey.